src/NzbDrone.Core/Download/Aggregation/Aggregators/AggregateLanguages.cs

```diff
@@ -76,9 +76,19 @@ public RemoteEpisode Aggregate(RemoteEpisode remoteEpisode)
                 languages = languages.Except(languagesToRemove).ToList();
             }
 
-            if ((languages.Count == 0 || (languages.Count == 1 && languages.First() == Language.Unknown)) && releaseInfo is { IndexerId: > 0 } && releaseInfo.Title.IsNotNullOrWhiteSpace())
+            if ((languages.Count == 0 || (languages.Count == 1 && languages.First() == Language.Unknown)) &&
+                releaseInfo?.Title?.IsNotNullOrWhiteSpace() == true &&
+                (releaseInfo is { IndexerId: > 0 } || releaseInfo.Indexer?.IsNotNullOrWhiteSpace() == true))
             {
-                var indexer = _indexerFactory.Get(releaseInfo.IndexerId);
+                IndexerDefinition indexer;
+                if (releaseInfo is { IndexerId: > 0 })
+                {
+                    indexer = _indexerFactory.Get(releaseInfo.IndexerId);
+                }
+                else
+                {
+                    indexer = _indexerFactory.All().FirstOrDefault(v => v.Name.EqualsIgnoreCase(releaseInfo.Indexer));
```

Comment:
Instead of getting all indexers then filtering by name we should be able to add a lookup in the DB by name, which could be made case sensitive.